src/components/button/nz-button.component.ts

```diff
@@ -40,7 +40,12 @@ export class NzButtonComponent implements AfterContentInit {
 
   @Input()
   set nzGhost(value: boolean) {
-    this._ghost = value;
+    const ghost = value as boolean | ''
```

Comment:
This way it will not change the property type, user still get `boolean` value when access via API.